lib/base/sysloglogger.cpp

```diff
@@ -61,6 +62,44 @@ void SyslogLogger::StaticInitialize()
 	m_FacilityMap["LOG_UUCP"] = LOG_UUCP;
 }
 
+bool SyslogHelper::ValidateFacility(const String& facility)
+{
+	if (m_FacilityMap.find(facility) == m_FacilityMap.end()) {
+		try {
+			Convert::ToLong(facility);
+		} catch (const std::exception&) {
+			return false;
+		}
+	}
+	return true;
+}
+
+int SyslogHelper::SeverityToNumber(LogSeverity severity)
+{
+	switch (severity) {
+	case LogDebug:
+		return LOG_DEBUG;
+	case LogNotice:
+		return LOG_NOTICE;
+	case LogWarning:
+		return LOG_WARNING;
+	case LogCritical:
+		return LOG_CRIT;
+	case LogInformation:
+	default:
+		return LOG_INFO;
```

Comment:
If you carefully compare https://github.com/Icinga/icinga2/blob/95cdc00ad4e1ab788794322c92c54cd3f5c8c2bf/lib/base/logger.hpp#L19-L26 with https://datatracker.ietf.org/doc/html/rfc5424#section-6.2.1, you'll notice that the order of notice and information is different, so while notice is a lower priority than information in icinga2, that message will have a higher priority in the journal (only noticed that because `journalctl` displayed the less important messages in bold, which looked weird).

So I think, `LogNotice` should map to `LOG_INFO` and `LogInfo` to `LOG_NOTICE` (great, not confusing at all). When changing this here, it will also affect `SyslogLogger`, but I think that's fine as the current behavior there is also broken and if anyone actually relied on filtering based on priorities in syslog, they should have noticed and opened a bug (nobody did as far as I can tell) or their filtering is broken already anyways. But when changing this, this needs a notice here in the code (to prevent someone from thinking "oh, there probably was a copy/paste error here, let's fix it" sometime in the future) and in die documentation for `JournaldLogger` and `SyslogLogger` (hoping that might prevent some confusion when someone runs `journalctl -p notice`).

But before changing this for `SyslogLogger`, I'd like a second opinion here, @lippserd @N-o-X (please hit :+1: if you agree that changing `SyslogLogger` is fine).

Reply:
> LogNotice should map to LOG_INFO and LogInfo to LOG_NOTICE

Imo sounds a bit too confusing. How about switching the meaning of `LogNotice` and `LogInformation` by changing order in `enum LogSeverity`, and replace usages in `Log()` with the counterpart to keep message importance as before? After that we align with what the rest of the world (=syslog) thinks of Notice and Information.

Drawback: Existing setups with `severity = "notice"` will suddenly see a bit less logs than before after their next icinga2 update, and existing configurations with `severity = "information"` will suddenly see a bit more messages. But sounds like the less of two evils, as really important messages (>= warning) would not be affected.

Reply:
> After that we align with what the rest of the world (=syslog) thinks of Notice and Information.

This would be nice if we were starting from scratch, but

> and existing configurations with `severity = "information"` will suddenly see a bit more messages

is an understatement, unfortunately. The current `notice` level is already quite spammy and I think this has a too high risk of blowing up in large clusters.

Reply:
>> Maybe it deserves an own issue and own PR?

> Sounds good to me. Should help to not forget mentioning it in the changelog then.

Fix will be done separately, not in this PR.

Reply:
I've created #9012 to track that issue.